rfcs/text/0031-risk-fields.md

```diff
@@ -31,13 +43,74 @@ Stage 2: Add or update all remaining field definitions. The list should now be e
+
+Furthermore, these `risk` fields will provide users with an additional vector to filter, sort and correlate information within the Security solution. For example, users will be able to start investigations by running queries like the following:
+* "Show me the most critical and high-risk Windows hosts in my environment"
+* "Show me the activity that contributed towards making Host X high-risk"
+* "Show me how the risk of Host X changed over time"
+* "Show me Critical and high-risk users on Host X"
+
 
 ## Source data
 
 <!--
 Stage 1: Provide a high-level description of example sources of data. This does not yet need to be a concrete example of a source document, but instead can simply describe a potential source (e.g. nginx access log). This will ultimately be fleshed out to include literal source examples in a future stage. The goal here is to identify practical sources for these fields in the real world. ~1-3 sentences or unordered list.
 -->
-
+The Host and User Risk Score views in the Security solution are/will be backed by indices produced as a result of running [transforms](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-transform.html) on a variety of data sources, not limited to the alerts data streams. An example document produced by the Host Risk Score transform in the absence of ECS `risk` fields looks as follows:
+
+```
+{
+  "risk_stats": {
+    "rule_risks": [
+      {
+        "rule_id": "c7ce36c0-32ff-4f9a-bfc2-dcb242bf99f9",
+        "rule_name": "Unusual File Modification by dns.exe",
+        "rule_risk": 73
+      },
+      {
+        "rule_id": "343a0b6b-71bb-4a8c-a7a7-afcdfc4df019",
+        "rule_name": "Unusual Windows Path Activity [Duplicate]",
+        "rule_risk": 2.4057357238464423
+      },
+      {
+        "rule_id": "43b68103-7eda-4687-9c0d-31308837a082",
+        "rule_name": "Anomalous Windows Process Creation [Duplicate]",
+        "rule_risk": 2.4057357238464423
+      },
+      {
+        "rule_id": "99cb72b7-ab55-4375-8bbe-696e3e1b8fe2",
+        "rule_name": "Unusual Windows Path Activity",
+        "rule_risk": 2.4057357238464423
+      },
+      {
+        "rule_id": "c3324b51-6d64-47e6-b220-2daa421e0468",
+        "rule_name": "Unusual Process For a Windows Host [Duplicate]",
+        "rule_risk": 21
+      }
+    ],
+    "risk_score": 78.61701409613882,
+    "risk_multipliers": [
+      "Host is a server",
+      "Tactic TA0001"
+    ]
+  },
+  "@timestamp": "2022-01-27T15:43:01.204Z",
+  "host": {
+    "name": "Host X"
+  },
+  "ingest_timestamp": "2022-01-27T15:49:00.335519179Z",
+  "risk": "High"
+}
+```
+
+With the introduction of ECS `risk` fields, fields in the above document would map as follows:
+* `risk_stats.rule_risks` -> `host.risk.info.rule_risks`
+* `risk_stats.risk_score` -> `host.risk.score`
+* `risk_stats.risk_multipliers` -> `host.risk.info.risk_multipliers`
```

Comment:
Could you add `risk_multipliers` to the field additions section above?

Reply:
I had a question about `risk_multipliers` and `rule_risks` actually. So these fields might not apply to all types of assets that have a risk associated with them. For example, `rule_risks` is supposed to provide information about the rules that contributed to the risk of a particular host. This might not apply to say events. Is there a better naming terminology for these?

Reply:
Better name in the sense that they are 'optional' since they don't apply to certain things?

Reply:
Exactly!

Reply:
I think keeping their names as you have described here makes sense. And then in the description you note the 'optional-ness' alongside maybe some helpful examples.

Reply:
@kgeller Done! (Hopefully)